src/org/webonise/multithreading/waitandnotify/Application.java

```diff
@@ -0,0 +1,22 @@
+package org.webonise.multithreading.waitandnotify;
+
+public class Application {
+
+    public void start(){
+
+        Thread thread = new Thread(new WorkerThread());
+        System.out.println("In main thread. Launching worker thread ...");
+        thread.start();
+
+        synchronized (thread){
```

Comment:
Yes that may cause problem. Now I have created a separate model for the shared item and main thread and worker thread communicate via wait and notify on this shared item object.